flake.nix

```diff
@@ -0,0 +1,28 @@
+{
```

Comment:
May you leave instructions on top of this file, as comments, about to (re-)generate the flakes and other nix artefacts. 

If we're going to maintain this, we'll need to know how ☺️

Reply:
Added 👍.

Flake updates should be in sync with package updates so as long as there's consideration whenever Cargo.toml changes. CI is also possible (and fairly simple, just invoke the nix command, diff that vs current).

There may be a cargo-native solution for caching, but we could do this pretty easily just taking advantage of nix. [premium nix service] has a free 5GB plan which is more than enough for all of the rust crates a few times over. I notice currently CI lags.